Assets/Scripts/Utilities/PartitionsExtensions.cs

```diff
@@ -0,0 +1,70 @@
+using System.Collections.Generic;
+using System.Linq;
+
+using Maes.Map;
+using Maes.Map.Generators.Patrolling.Partitioning;
+
+namespace Maes.Utilities
+{
+    public static class PartitionsExtensions
+    {
+        public struct PossibleMeetingPoint
+        {
+            public readonly int Robot1Id;
+            public readonly int Robot2Id;
+            public readonly List<(int, int)> Connections;
+
+            public PossibleMeetingPoint(int robot1Id, int robot2Id)
+            {
+                Robot1Id = robot1Id;
+                Robot2Id = robot2Id;
+                Connections = new List<(int, int)>();
+            }
+        }
+
+        public static List<PossibleMeetingPoint> GetNeighborsPartitionsWithNoCommonVertices(this Dictionary<int, PartitionInfo> partitionInfoByRobotId, PatrollingMap patrollingMap)
+        {
+            var possibleMeetingPoints = new List<PossibleMeetingPoint>();
+
+            var partitions = partitionInfoByRobotId.ToArray();
+
+            foreach (var (robot1Id, partitionInfo1) in partitions)
+            {
+                foreach (var (robot2Id, partitionInfo2) in partitions.Skip(robot1Id + 1))
+                {
+                    var commonVerticesCount = partitionInfo1.VertexIds.Intersect(partitionInfo2.VertexIds).Count();
+                    if (commonVerticesCount > 0)
+                    {
+                        continue;
+                    }
+
+                    var possibleMeetingPoint = new PossibleMeetingPoint(robot1Id, robot2Id);
```

Comment:
4 Foreach deep.. maybe extract a local method called something like: 
PartitionInfo FindMeetingPointsForPartitions(map, partition1, partition2){...}
Also this makes testing easier ;)